src/lib/externalAttachment/index.ts

```diff
@@ -0,0 +1,127 @@
+import { singleton } from 'tsyringe'
+import { Logger } from 'pino'
+import { logger } from '../logger.js'
+import Identity from '../identity.js'
+import env from '../../env.js'
+import { AttachmentRow } from '../db/types.js'
+import { z } from 'zod'
+import contentDisposition from 'content-disposition'
+import * as fs from 'fs'
+import * as path from 'path'
+
+const OidcConfigSchema = z.object({
+  token_endpoint: z.string().url(),
+})
+
+const AccessTokenResponseSchema = z.object({
+  access_token: z.string(),
+})
+
+type OidcConfig = z.infer<typeof OidcConfigSchema>
+
+interface Credential {
+  username: string
+  secret: string
+  owner: string
+}
+
+function loadCredentials(): Credential[] {
+  const credentialsPath = path.resolve(process.cwd(), env.CREDENTIALS_FILE_PATH)
+  const rawData = fs.readFileSync(credentialsPath, 'utf-8')
+  const parsed = JSON.parse(rawData)
+  return parsed.credentials
+}
```

Comment:
we shoulod make this more robust. Move this to a new class and dependency inject it into this class. This code also synchronously loads the file and loads it from scratch on every credential fetch. So you need to make sure:
* the credentials file is valid, use `Zod`
* that you memoize the result of parsing so we only load and validate it once
* you handle file system errors logging out appropriately. It doesn't matter if the process exits but we need an error log explaining why the load or parse failed
* it might be worth converting it into a map data structure so you can fetch credentials for a given address